### PR TITLE
Feature/uppsf 2838 configured ontology

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**
# Description

## What

Merge ontology changes from the development branch. All features have been tested for at least a week on the testing pipeline. 

## Why

Introduction to the dynamically configurable ontology.

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: ontology/aggregate_test.go [hunk 1]
```diff
@@ -37,6 +49,89 @@ func TestCreateAggregateConcept(t *testing.T) {
 	}
 }
 
+func TestCreateAggregateConcept_Properties(t *testing.T) {
```
Comment (ivanruski) [inline ontology/aggregate_test.go:52]: **question:** Do we have a test case here or the test above where we have more properties than the ones we actually need? In other words the normalised store returned fields we don't want.

**Comment by ivanruski** [comment]
**nitpick (non-blocking|for-future-commit):** We can remove the nolint directive from `mergeCanonicalInformation` as it's complexity is 10 now.

**Comment by ivanruski** [comment]
**question:** In `config.yml` for the properties we use `neoProp` and for the relationships we use `conceptField`. Why is that ?

**Review by ivanruski — APPROVED** [review]
LGTM. Just a couple of questions to clarify things for myself.

**Comment by ivan-p-nikolov** [comment]
> **question:** In `config.yml` for the properties we use `neoProp` and for the relationships we use `conceptField`. Why is that ?

`neoProp` is used by `concepts-rw-neo4j` service. It specifies what is the name of the property stored in the Neo4j node. In the current setup we use the key in the property map for `conceptField`. For relationships we've implemented it the other way around. The map key is the relationship label stored in Neo4j.

I haven't decided on a correct approach so I've left it for later. So when we extract the ontology as a lib then we can see what works best.

File: ontology/aggregate_test.go (hunk 1 above)
Comment (ivan-p-nikolov) [reply ontology/aggregate_test.go:52]: Not sure on what you think it should be tested.

**Comment by ivanruski** [comment]
> > **question:** In `config.yml` for the properties we use `neoProp` and for the relationships we use `conceptField`. Why is that ?
> 
> `neoProp` is used by `concepts-rw-neo4j` service. It specifies what is the name of the property stored in the Neo4j node. In the current setup we use the key in the property map for `conceptField`. For relationships we've implemented it the other way around. The map key is the relationship label stored in Neo4j.
> 
> I haven't decided on a correct approach so I've left it for later. So when we extract the ontology as a lib then we can see what works best.

Yep, you're right, I think I get it now.

File: ontology/aggregate_test.go (hunk 1 above)
Comment (ivanruski) [reply ontology/aggregate_test.go:52]: Maybe there is no need but I was thinking of test where we have a concept in S3 looking like:
```json{
  "uuid": "77e5a886-f21b-3001-b380-d44c11f603ce",
  "type": "Organisation",
  "PropertyWhichIsNotPresentInConfig.yaml" : "value",
  ...other valid props
```
And in this case we expect that the resulted aggregated concept won't have the property.

Comment (ivan-p-nikolov) [reply ontology/aggregate_test.go:52]: I see this as a regression test. Meaning that it tries to cover the old functionality and confirm that it still works. 
There are other tests that cover that a property not defined in the config will not be stored in the concept.

**Review by epavlova — APPROVED** [review]
**thought for the future:** Do you think that it could be beneficial to have the relationship properties in the config description of a relationship. For example to have in the config that `HAS_INDUSTRY_CLASSIFICATION` have field `rank`.

**Comment by ivan-p-nikolov** [comment]
> **thought for the future:** Do you think that it could be beneficial to have the relationship properties in the config description of a relationship. For example to have in the config that `HAS_INDUSTRY_CLASSIFICATION` has field `rank`.

Yes, it was my plan to have relationship properties as well, but didn't see the need for now to explicitly add them in the config.

**Pull request merged**